lib/constructSignatureString.test.js

```diff
@@ -37,6 +37,27 @@ test('Constructs a signature string from the incoming request', async ({
         getDigest: () => 'base64'
       },
       expected: defaultSignature
+    },
+    {
+      description:
+        'Matching signature constructed when a valid request using a digest header is provided',
+      input: {
+        headers: {
+          signature: 'keyId="key-a", headers="(request-target) (created) (expires) host digest content-type", signature="pQul5YFrqv76Zq2bE1kWjJfFGnTu0MwU7X7c8MWDswAI5V7dROqKbBWKUGcoysxujgTqkJo/Eg74x34o54hqRg==", created=1402170695, expires=1402170895',
+          host: 'localhost:3000',
+          digest: 'sha-512=+PtokCNHosgo04ww4cNhd4yJxhMjLzWjDAKtKwQZDT4Ef9v/PrS/+BQLX4IX5dZkUMK/tQo7Uyc68RkhNyCZVg==',
+          'content-type': 'application/json'
+        },
+        body: { hello: 'world' },
+        raw: { method: 'POST', url: '/' }
+      },
+      options: {
+        ...defaultOptions,
+        sharedSecret: 'topSecret',
+        getAlgorithm: () => 'sha512',
+        getDigest: () => 'base64'
```

Comment:
Should this method be `getDigestEncoding`?

Reply:
I agree and discussed it with Ted. I was planning on making this change when I tackle issue #3 . Happy to do it here but as well but it fits in better with that work.

Reply:
I renamed this `getDigest` to `getHMACEncoding` to differentiate it from the newly created `digestEncoding` option. The previous `constructBodyDigestSignature` file and function has been renamed to to `getDigest`.